app/controllers/users/invitations_controller.rb

```diff
@@ -20,15 +20,19 @@ class Users::InvitationsController < Devise::InvitationsController
   def create
     super do |invited_user|
       # set default values
-      invited_user.update(role: invited_user.role || "agent" )
+      invited_user.update(role: invited_user.role || "agent")
     end
   end
 
   private
 
   # Override superclass method for default params for newly created invites, allowing us to add attributes
   def invite_params
-    params.require(:user).permit(:name, :email, :vita_partner_id)
+    vita_partner_id = params.require(:user).require(:vita_partner_id)
+    vita_partner = VitaPartner.find(vita_partner_id)
+    if Ability.new(current_user).can?(:manage, vita_partner)
+      params.require(:user).permit(:name, :email, :vita_partner_id)
+    end
```

Comment:
I'm surprised to see queries and ability checks in the `_params` method. Ordinarily these are handled elsewhere.

In general we should never be using Ability.new directly unless we are creating separate Ability classes. `Ability.new( current_user)` is always available as `#current_ability` in any controller that inherits from `ApplicationController`

Reply:
This impliese that we would permit different params depending on a user's permissions